shared/test/dataDriver_test.js

```diff
@@ -0,0 +1,222 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+describe("loop.DataDriver", () => {
+  "use strict";
+
+  let { expect } = chai;
+  let { actions } = loop.shared;
+  let { CHAT_CONTENT_TYPES } = loop.shared.utils;
+
+  let clock, fakeXHR, requests, sandbox;
+  let dispatcher, driver;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    sandbox = LoopMochaUtils.createSandbox();
+
+    dispatcher = new loop.Dispatcher();
+    sandbox.stub(dispatcher, "dispatch");
+
+    driver = new loop.DataDriver({
+      dispatcher
+    });
+
+    fakeXHR = sandbox.useFakeXMLHttpRequest();
+    requests = [];
+    fakeXHR.xhr.onCreate = function(xhr) {
+      requests.push(xhr);
+    };
+    sandbox.stub(window, "EventSource", url => {
+      requests.push({ url });
+      return { addEventListener: () => {} };
+    });
+  });
+
+  afterEach(() => {
+    clock.restore();
+    sandbox.restore();
+    LoopMochaUtils.restore();
+  });
+
+  describe("Constructor", () => {
+    it("should throw an error if the dispatcher is missing", () => {
+      expect(() => new loop.DataDriver({})).to.Throw(/dispatcher/);
+    });
+  });
+
+  describe("#sendTextChatMessage", () => {
+    it("should send a message by updating a chat record", () => {
+      driver.sendTextChatMessage({
+        contentType: CHAT_CONTENT_TYPES.TEXT,
+        message: "Are you there?"
+      });
+
+      let { method, requestBody, url } = requests[0];
+      expect(method).eql("PUT");
+      expect(requestBody).eql('{"timestamp":{".sv":"timestamp"},"value":{"contentType":"chat-text","message":"Are you there?"}}');
+      expect(url.match(/([^\/]+).{12}\.json$/)[1]).eql("chat!00000000");
+    });
+  });
+
+  describe("#fetchServerData", () => {
+    it("should connect to the room on server data", () => {
```

Comment:
Somewhere we need to add tests for doing the initial connect/update, getting the clock skew and then doing the initial requestChat.